### PR TITLE
API "Source" to "Endpoint"

### PR DESCRIPTION
**Pull request by Akvannortwick**
Correcting README file:
Source to Endpoint as mentioned in #682

Still working through other docs.

### CODE REVIEW

File: docs/api/fluview.md [hunk 1]
```diff
@@ -5,18 +5,18 @@ parent: Epidata API (Other Diseases)
...
+General topics not specific to any particular endpoint are discussed in the
 [API overview](README.md). Such topics include:
 [contributing](README.md#contributing), [citing](README.md#citing), and
 [data licensing](README.md#data-licensing).
 
 ## FluView Data
 
 Influenza-like illness (ILI) from U.S. Outpatient Influenza-like Illness Surveillance Network (ILINet).
- - Data source: [United States Centers for Disease Control and Prevention](http://gis.cdc.gov/grasp/fluview/fluportaldashboard.html) (CDC)
+ - Endpoint: [United States Centers for Disease Control and Prevention](http://gis.cdc.gov/grasp/fluview/fluportaldashboard.html) (CDC)
```
Comment (krivard) [inline docs/api/fluview.md:19]: (whoops, this one is a citation)

```suggestion
 - Data source: [United States Centers for Disease Control and Prevention](http://gis.cdc.gov/grasp/fluview/fluportaldashboard.html) (CDC)
```

File: docs/api/gft.md [hunk 2]
```diff
@@ -5,18 +5,18 @@ parent: Epidata API (Other Diseases)
...
 [API overview](README.md). Such topics include:
 [contributing](README.md#contributing), [citing](README.md#citing), and
 [data licensing](README.md#data-licensing).
 
 ## Google Flu Trends Data
 
-Estimate of influenza activity based on volume of certain search queries. Google has discontinued Flu Trends, and this is now a static data source.
- - Data source: [Google](https://www.google.org/flutrends/)
+Estimate of influenza activity based on volume of certain search queries. Google has discontinued Flu Trends, and this is now a static endpoint.
+ - Endpoint: [Google](https://www.google.org/flutrends/)
```
Comment (krivard) [inline docs/api/gft.md:19]: (more specific, and another citation)

```suggestion
Estimate of influenza activity based on volume of certain search queries. Google has discontinued Flu Trends, and this is now a static dataset.
 - Data source: [Google](https://www.google.org/flutrends/)
```

File: docs/api/nidss_dengue.md [hunk 3]
```diff
@@ -5,18 +5,18 @@ parent: Epidata API (Other Diseases)
...
+General topics not specific to any particular endpoint are discussed in the
 [API overview](README.md). Such topics include:
 [contributing](README.md#contributing), [citing](README.md#citing), and
 [data licensing](README.md#data-licensing).
 
 ## NIDSS Dengue Data
 
 Counts of confirmed dengue cases from Taiwan's NIDSS.
- - Data source: [Taiwan CDC](http://nidss.cdc.gov.tw/en/SingleDisease.aspx?dc=1&dt=4&disease=061&position=1)
+ - Endpoint: [Taiwan CDC](http://nidss.cdc.gov.tw/en/SingleDisease.aspx?dc=1&dt=4&disease=061&position=1)
```
Comment (krivard) [inline docs/api/nidss_dengue.md:19]:
```suggestion
 - Data source: [Taiwan CDC](http://nidss.cdc.gov.tw/en/SingleDisease.aspx?dc=1&dt=4&disease=061&position=1)
```

File: docs/api/nidss_flu.md [hunk 4]
```diff
@@ -5,18 +5,18 @@ parent: Epidata API (Other Diseases)
...
+General topics not specific to any particular endpoint are discussed in the
 [API overview](README.md). Such topics include:
 [contributing](README.md#contributing), [citing](README.md#citing), and
 [data licensing](README.md#data-licensing).
 
 ## NIDSS Flu Data
 
 Outpatient ILI from Taiwan's National Infectious Disease Statistics System (NIDSS).
- - Data source: [Taiwan CDC](http://nidss.cdc.gov.tw/en/CDCWNH01.aspx?dc=wnh)
+ - Endpoint: [Taiwan CDC](http://nidss.cdc.gov.tw/en/CDCWNH01.aspx?dc=wnh)
```
Comment (krivard) [inline docs/api/nidss_flu.md:19]:
```suggestion
 - Data source: [Taiwan CDC](http://nidss.cdc.gov.tw/en/CDCWNH01.aspx?dc=wnh)
```

File: docs/api/nowcast.md [hunk 5]
```diff
@@ -5,18 +5,18 @@ parent: Epidata API (Other Diseases)
...
+General topics not specific to any particular endpoint are discussed in the
 [API overview](README.md). Such topics include:
 [contributing](README.md#contributing), [citing](README.md#citing), and
 [data licensing](README.md#data-licensing).
 
 ## ILI Nearby Data
 
 A nowcast of U.S. national, regional, and state-level (weighted) %ILI, available seven days (regionally) or five days (state-level) before the first ILINet report for the corresponding week.
- - Data source: [Delphi's ILI Nearby system](https://delphi.cmu.edu/nowcast/)
+ - Endpoint: [Delphi's ILI Nearby system](https://delphi.cmu.edu/nowcast/)
```
Comment (krivard) [inline docs/api/nowcast.md:19]:
```suggestion
 - Data source: [Delphi's ILI Nearby system](https://delphi.cmu.edu/nowcast/)
```

File: docs/api/sensors.md [hunk 6]
```diff
@@ -26,7 +26,7 @@ Trends, etc., during the COVID-19 pandemic, because these were designed to track
 ILI as driven by seasonal influenza, and were NOT designed to track ILI during
 the COVID-19 pandemic.
 
-We are working on collecting new ILI-related data sources may be useful to
+We are working on collecting new ILI-related endpoints may be useful to
 nowcasting and forecasting ILI during the COVID-19 pandemic, and will make these
 available as soon as possible.
```
Comment (krivard) [inline docs/api/sensors.md:31]: @RoniRos is this paragraph relevant anymore? if not we should delete it

```suggestion
```

File: docs/api/twitter.md [hunk 7]
```diff
@@ -5,18 +5,18 @@ parent: Epidata API (Other Diseases)
...
+General topics not specific to any particular endpoint are discussed in the
 [API overview](README.md). Such topics include:
 [contributing](README.md#contributing), [citing](README.md#citing), and
 [data licensing](README.md#data-licensing).
 
 ## Twitter Stream Data
 
 Estimate of influenza activity based on analysis of language used in tweets.
- - Data source: [HealthTweets](http://www.healthtweets.org/)
+ - Endpoint: [HealthTweets](http://www.healthtweets.org/)
```
Comment (krivard) [inline docs/api/twitter.md:19]:
```suggestion
 - Data source: [HealthTweets](http://www.healthtweets.org/)
```

File: docs/api/wiki.md [hunk 8]
```diff
@@ -5,18 +5,18 @@ parent: Epidata API (Other Diseases)
...
+General topics not specific to any particular endpoint are discussed in the
 [API overview](README.md). Such topics include:
 [contributing](README.md#contributing), [citing](README.md#citing), and
 [data licensing](README.md#data-licensing).
 
 ## Wikipedia Access Data
 
 Number of page visits for selected English, Influenza-related wikipedia articles.
- - Data source: [Wikimedia](https://dumps.wikimedia.org/other/pagecounts-raw/)
+ - Endpoint: [Wikimedia](https://dumps.wikimedia.org/other/pagecounts-raw/)
```
Comment (krivard) [inline docs/api/wiki.md:19]:
```suggestion
 - Data source: [Wikimedia](https://dumps.wikimedia.org/other/pagecounts-raw/)
```

**Review by krivard — CHANGES_REQUESTED** [review]
The citation changes can go in, but let's wait for roni's feedback on the "we are working on ILI-related stuff" paragraph before doing anything about it

File: docs/api/gft.md (hunk 2 above)
Comment (Akvannortwick) [reply docs/api/gft.md:19]:
```suggestion
Estimate of influenza activity based on volume of certain search queries. Google has discontinued Flu Trends, and this is now a static endpoint.
 - Source: [Google](https://www.google.org/flutrends/)
```

File: docs/api/nidss_flu.md (hunk 4 above)
Comment (Akvannortwick) [reply docs/api/nidss_flu.md:19]:
```suggestion
 - Source: [Taiwan CDC](http://nidss.cdc.gov.tw/en/CDCWNH01.aspx?dc=wnh)
```

File: docs/api/nowcast.md (hunk 5 above)
Comment (Akvannortwick) [reply docs/api/nowcast.md:19]:
```suggestion
 - Source: [Delphi's ILI Nearby system](https://delphi.cmu.edu/nowcast/)
```

File: docs/api/wiki.md (hunk 8 above)
Comment (Akvannortwick) [reply docs/api/wiki.md:19]:
```suggestion
 - Source: [Wikimedia](https://dumps.wikimedia.org/other/pagecounts-raw/)
```

File: docs/api/sensors.md (hunk 6 above)
Comment (RoniRos) [reply docs/api/sensors.md:31]: No longer relevant, and should be deleted.  (Also "Data sources" was not used to mean 'endpoints'.)

Also, @krivard : I noticed both above and below numerous (in fact, most!) cases where 'data source' was replaced by 'endpoint' even though that was not the intended meaning of that term - it should be left as 'data source'.   This substitution should not be done automatically!

Also:

> This is the documentation of the API for accessing the Delphi's Digital Surveillance Sensors (`sensors`) endpoint of
the [Delphi](https://delphi.cmu.edu/)'s epidemiological data. 

This is garbled and overly wordy.  I suspect it should be something like:

"This is the documentation of the API for accessing Delphi's Epidata -- epidemiological sensors and indicators database. "

**Review by RoniRos — CHANGES_REQUESTED** [review]
Most original occurrences of 'data source' in this document do not mean 'endpoint'!  They should be left as is.  AFAIK, the only endpoints currently in Epidata are:
 - COVIDcast
 - The Survey
 - The HHS hospital utilization & capacity reports
 - Flu & other diseases (the only one from before 2020)

**Comment by krivard** [comment]
@RoniRos This PR changes the name of the API parameter called `source` to be `endpoint` as described in #321.

We now define an "endpoint" as an API query that accepts a distinct set of parameters, keyed by the value of the `endpoint` or `source` parameter (the old name is also accepted, for backwards-compatibility). Since each Epidata dataset is stored in its own database table, and has its own dedicated PHP subroutine for handling it, its query signature may be changed independently of the others. This means that we now have one endpoint for each individual Epidata dataset.

Perhaps you were not be aware of the full extent of this change -- the full intended list of all such endpoints is below.

* `endpoint=covidcast`
* `endpoint=fb-survey`
* `endpoint=covidcast_meta`
* `endpoint=covid_hosp_facility_lookup`
* `endpoint=covid_hosp_facility`
* `endpoint=covid_hosp`
* `endpoint=delphi`
* `endpoint=flusurv`
* `endpoint=fluview_clinical`
* `endpoint=fluview`
* `endpoint=fluview_meta`
* `endpoint=gft`
* `endpoint=meta`
* `endpoint=nidss_dengue`
* `endpoint=nidss_flu`
* `endpoint=nowcast`
* `endpoint=fluview`
* `endpoint=wiki`
* `endpoint=sensors`
* `endpoint=twitter`
* `endpoint=wiki`
* `endpoint=fluview`
* `endpoint=fluview_meta`

**Comment by krivard** [comment]
@RoniRos please confirm -- your comment

> This is garbled and overly wordy. I suspect it should be something like:

refers to this page: https://cmu-delphi.github.io/delphi-epidata/api/sensors.html

?

"Delphi's Epidata -- epidemiological sensors and indicators database. " seems awfully general for that.

**Comment by RoniRos** [comment]
> Perhaps you were not be aware of the full extent of this change -- the full intended list of all such endpoints is below.

I was indeed unaware, and still don't understand why this is necessary.  Can the two of us has in out "in person"?

File: docs/api/sensors.md [hunk 9]
```diff
@@ -5,10 +5,10 @@ parent: Epidata API (Other Diseases)
 
 # Digital Surveillance Sensors
 
-This is the documentation of the API for accessing the Delphi's Digital Surveillance Sensors (`sensors`) data source of
+This is the documentation of the API for accessing the Delphi's Digital Surveillance Sensors (`sensors`) endpoint of
```
Comment (krivard) [inline docs/api/sensors.md:8]:
```suggestion
This is the documentation of the API for accessing the Digital Surveillance Sensors (`sensors`) endpoint of
```

**Review by krivard — APPROVED** [review]
j'approve

**Pull request merged**